upgrade_test.py

```diff
@@ -517,7 +517,7 @@ def test_rolling_upgrade(self):  # pylint: disable=too-many-locals,too-many-stat
         well, upgrade all nodes to new version in the end.
         """
         # In case the target version >= 3.1 we need to perform test for truncate entries
-        target_upgrade_version = self.params.get('target_upgrade_version')
+        target_upgrade_version = self.params.get('target_upgrade_version').replace('~', '-')
```

Comment:
We need a more global fix for this.

every place we are using parse_version might be broken by this, since the legacy version support is now deprecated and removed.

we will see much more of those happening.

(I've faced it also in CCM)

Reply:
i have a task for that --> https://github.com/scylladb/qa-tasks/issues/39